### PR TITLE
Add support for sf ^6.0

### LINKED ISSUE #256: Symfony 6.0 compatibility
### Bug Report
|------------ | ------
| BC Break    | No
| Version     | 4.2.0

#### Summary

Upgrade to Symfony 5.4 works perfectly well. upgrading to the next major version Symfony 6.0 results in unresolvable set of packages.

Problem 1
    - oneup/flysystem-bundle[4.0.0, ..., 4.2.0] require symfony/config ^4.4 || ^5.0 -> found symfony/config[v4.4.0, ..., v4.4.34, v5.0.0, ..., v5.4.0] but these were not loaded, likely because it conflicts with another require.
    - Root composer.json requires oneup/flysystem-bundle ^4.0 -> satisfiable by oneup/flysystem-bundle[4.0.0, ..., 4.2.0].

#### How to reproduce

Set up a brand new symfony project:
run the follwing commands in the console:

1. symfony new flyupsix_point_o
2. cd flyupsix_point_o
3. composer require oneup/flysystem-bundle -W

this will result in: 

Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires oneup/flysystem-bundle ^4.2 -> satisfiable by oneup/flysystem-bundle[4.2.0].
    - oneup/flysystem-bundle 4.2.0 requires symfony/config ^4.4 || ^5.0 -> found symfony/config[v4.4.0, ..., v4.4.34, v5.0.0, ..., v5.4.0] but the package is fixed to v6.0.0 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.

### PR DESCRIPTION
**Pull request by odolbeau**
This PR:
* Add support for symfony ^6.0
* Remove support for symfony >5.0 & <5.3
* Remove PHPStan & php-cs-fixer from dev dependencies (it uses `jakzal/phpqa` docker image instead)
* Add a Makefile in order to quickly run CS / tests.

Last 2 points can be reverted if needed.

Fix #256

### CODE REVIEW

**Comment by odolbeau** [comment]
If (as I am), you're bored of approving workflows run, I just found this can be configured [here](https://github.com/organizations/1up-lab/settings/actions) as explained [in the documentation](https://docs.github.com/en/organizations/managing-organization-settings/disabling-or-limiting-github-actions-for-your-organization#configuring-required-approval-for-workflows-from-public-forks)! :partying_face:

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -9,35 +9,18 @@ on:
...
     tests:
         name: PHP ${{ matrix.php }} / SF ^${{ matrix.symfony }}
         runs-on: ubuntu-latest
         strategy:
             fail-fast: false
             matrix:
-                php: [7.4, 8.0]
-                symfony: [4.4, 5.0]
+                php: [7.4, 8.0, 8.1]
+                symfony: [4.4, 5.3, 6.0]
```
Comment (metfan) [inline .github/workflows/ci.yml:19]: Why not using sf 5.4 instead of 5.3 ?

Comment (bytehead) [reply .github/workflows/ci.yml:19]: I would also go for 5.4 👍

**Comment by elkuku** [comment]
Please merge :wink:

**Review by bytehead — CHANGES_REQUESTED** [review]
Thank you very much @odolbeau 🙏 

Would you mind to make a seperate PR for your last 2 points? 

> - Remove PHPStan & php-cs-fixer from dev dependencies (it uses jakzal/phpqa docker image instead)
> - Add a Makefile in order to quickly run CS / tests.

**Comment by ltlsquare** [comment]
Looks good guys, is there anything else that needs to be done? can help out to get this over the line?

**Comment by odolbeau** [comment]
There is some permission errors with PHP 8 but I cannot reproduce the bug on my laptop.
@meyerbaptiste, it looks like its linked to the visibility config you added in 4.2.0, could you please take a look? :pray:

**Comment by meyerbaptiste** [comment]
@odolbeau, I just sent you a PR!

**Comment by odolbeau** [comment]
> @odolbeau, I just sent you a PR!

@meyerbaptiste do you have a link please? I don't know what you're talking about... :thinking:

**Comment by bytehead** [comment]
@odolbeau https://github.com/odolbeau/OneupFlysystemBundle/pull/1 🙃

**Comment by odolbeau** [comment]
Thanks! I didn't get notified, that's strange!
Anyway, thanks a lot @meyerbaptiste, it looks like CI is happy! :ok_hand:

**Review by bytehead — COMMENTED** [review]
As we test `4.4`, `5.4` and `6.0` in the CI, would it make sense to raise the requirements for `symfony/*` to `"^4.4|^5.4|^6.0"`? ~~The code from [`a62933e` (#257)](https://github.com/1up-lab/OneupFlysystemBundle/pull/257/commits/a62933e188ef0f7d6043a9d5f6fd3c2c3d19984d) is maybe obsolete then?~~ Is not, as we cover `4.4` as well.

**Comment by odolbeau** [comment]
I did not removed support for sf 5.3 as this version is stille maintained for now https://symfony.com/releases

**Review by bytehead — APPROVED** [review]

**Comment by bytehead** [comment]
Thank you @odolbeau & @meyerbaptiste ❤️

**Comment by bytehead** [comment]
I've released [4.3.0](https://github.com/1up-lab/OneupFlysystemBundle/releases/tag/4.3.0)

**Pull request merged**